cmake/flash/CMakeLists.txt

```diff
@@ -159,37 +159,33 @@ foreach(target flash debug debugserver attach)
     set(RUNNER_VERBOSE)
   endif()
 
-  # We pass --skip-rebuild here because the DEPENDS value ensures the
-  # build is already up to date before west is run.
   if(WEST)
-    set(cmd
-      ${CMAKE_COMMAND} -E env
-      ${WEST}
-      ${RUNNER_VERBOSE}
-      ${target}
-      --skip-rebuild
-      DEPENDS ${RUNNERS_DEPS}
-              $<TARGET_PROPERTY:zephyr_property_target,${TARGET_UPPER}_DEPENDENCIES>
-      WORKING_DIRECTORY ${APPLICATION_BINARY_DIR}
-      )
-
     add_custom_target(${target}
       COMMAND
-      ${cmd}
+        # This command will echo if <target> has added dependencies,
+	# else it will just be true and continue.
+        ${CMAKE_COMMAND}
+	  -E $<IF:$<BOOL:$<TARGET_PROPERTY:${target},MANUALLY_ADDED_DEPENDENCIES>>,echo,true>
+	  "ERROR: ${target} cannot have dependencies, please remove add_dependencies'('${target}"
+	  "$<TARGET_PROPERTY:${target},MANUALLY_ADDED_DEPENDENCIES>')' in build system."
+      COMMAND
+        ${CMAKE_COMMAND}
```

Comment:
Can you add a comment saying what this one does?

Reply:
Done.